recognition/advanced_tree_search.py

```diff
@@ -536,6 +537,151 @@ def create_config(
         return config, post_config, lm_gc_job
 
 
+class RescoreLatticeCacheJob(rasr.RasrCommand, Job):
+    def __init__(
+        self,
+        crp,
+        lattice_cache,
+        rescorer_type="single-best",
+        max_hypotheses=5,
+        pruning_threshold=16.0,
+        history_limit=0,
+        rescoring_lookahead_scale=1.0,
+        rtf: float = 1.0,
+        cpu: int = 1,
+        mem: float = 4.0,
+        use_gpu: bool = False,
+        extra_config=None,
+        extra_post_config=None,
+    ):
```

Comment:
as typical in @curufinwes Jobs, a docstring is missing xP
type hints would also be appreciated